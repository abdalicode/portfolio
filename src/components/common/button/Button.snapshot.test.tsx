import { render } from '@testing-library/react';
import React from 'react';

import { Button } from '.';

describe('Button Snapshot Tests', () => {
  const variants = [
    'default',
    'destructive',
    'outline',
    'secondary',
    'ghost',
    'link',
  ] as const;
  const sizes = ['default', 'sm', 'lg', 'icon'] as const;

  variants.forEach((variant) => {
    sizes.forEach((size) => {
      it(`renders correctly with variant: ${variant} and size: ${size}`, () => {
        const { asFragment } = render(<Button variant={variant} size={size} />);
        expect(asFragment()).toMatchSnapshot();
      });
    });
  });

  it('renders correctly as a child component', () => {
    const { asFragment } = render(
      <Button asChild>
        <span>Child Content</span>
      </Button>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
