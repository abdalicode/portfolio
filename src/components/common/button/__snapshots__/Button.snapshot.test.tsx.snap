// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot Tests renders correctly as a child component 1`] = `
<DocumentFragment>
  <span
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
  >
    Child Content
  </span>
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: default and size: default 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: default and size: icon 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 w-9"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: default and size: lg 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-10 rounded-md px-8"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: default and size: sm 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-8 rounded-md px-3 text-xs"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: destructive and size: default 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90 h-9 px-4 py-2"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: destructive and size: icon 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90 h-9 w-9"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: destructive and size: lg 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90 h-10 rounded-md px-8"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: destructive and size: sm 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90 h-8 rounded-md px-3 text-xs"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: ghost and size: default 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: ghost and size: icon 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 w-9"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: ghost and size: lg 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 rounded-md px-8"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: ghost and size: sm 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-8 rounded-md px-3 text-xs"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: link and size: default 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-9 px-4 py-2"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: link and size: icon 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-9 w-9"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: link and size: lg 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-10 rounded-md px-8"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: link and size: sm 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-8 rounded-md px-3 text-xs"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: outline and size: default 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 border border-input bg-transparent shadow-sm hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: outline and size: icon 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 border border-input bg-transparent shadow-sm hover:bg-accent hover:text-accent-foreground h-9 w-9"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: outline and size: lg 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 border border-input bg-transparent shadow-sm hover:bg-accent hover:text-accent-foreground h-10 rounded-md px-8"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: outline and size: sm 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 border border-input bg-transparent shadow-sm hover:bg-accent hover:text-accent-foreground h-8 rounded-md px-3 text-xs"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: secondary and size: default 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80 h-9 px-4 py-2"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: secondary and size: icon 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80 h-9 w-9"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: secondary and size: lg 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80 h-10 rounded-md px-8"
  />
</DocumentFragment>
`;

exports[`Button Snapshot Tests renders correctly with variant: secondary and size: sm 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80 h-8 rounded-md px-3 text-xs"
  />
</DocumentFragment>
`;
